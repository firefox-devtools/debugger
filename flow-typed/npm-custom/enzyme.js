/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// Using this custom definition of the enzyme module avoids generating some
// flow errors that are unrelated to uses of enzyme itself...
// It would be nice to clean this up.
declare module "enzyme" {
  declare module.exports: any;
}
