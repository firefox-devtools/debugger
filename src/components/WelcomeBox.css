.welcomebox {
  width: calc(100% - 1px);

  /* Offsetting it by 30px for the sources-header area */
  height: calc(100% - 30px);
  position: absolute;
  top: 30px;
  left: 0;
  padding: 50px 0 0 0;
  text-align: center;
  font-size: 1.25em;
  color: var(--theme-comment-alt);
  background-color: var(--theme-toolbar-background);
  font-weight: lighter;
  z-index: 100;
  user-select: none;
}

.theme-dark .welcomebox {
  background-color: var(--theme-body-background);
}

.alignlabel {
  display: inline-block;
}

.welcomebox .toggle-button-end {
  position: absolute;
  top: auto;
  bottom: 0;
  offset-inline-end: 0;
  offset-inline-start: auto;
}

.shortcutKeys {
  text-align: right;
  float: left;
  font-family: Courier;
}

.shortcutFunction {
  text-align: left;
  float: left;
  margin-left: 25px;
}

html .welcomebox .toggle-button-end.collapsed {
  bottom: 1px;
}
