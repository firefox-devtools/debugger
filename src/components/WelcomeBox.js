// @flow
import { DOM as dom, PropTypes, createClass, createFactory } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

import actions from "../actions";
import { getPaneCollapse } from "../selectors";
import { formatKeyShortcut } from "../utils/text";

const PaneToggleButton = createFactory(
  require("./shared/Button/PaneToggle").default
);

import "./WelcomeBox.css";

const WelcomeBox = createClass({
  propTypes: {
    horizontal: PropTypes.bool,
    togglePaneCollapse: PropTypes.func,
    endPanelCollapsed: PropTypes.bool
  },

  displayName: "WelcomeBox",

  renderToggleButton() {
    if (this.props.horizontal) {
      return;
    }

    return PaneToggleButton({
      position: "end",
      collapsed: !this.props.endPanelCollapsed,
      horizontal: this.props.horizontal,
      handleClick: this.props.togglePaneCollapse
    });
  },

  render() {
    const searchLabel = L10N.getFormatStr(
      "welcome.search",
      formatKeyShortcut(`CmdOrCtrl+${L10N.getStr("sources.search.key")}`)
    );
    return dom.div(
      { className: "welcomebox" },
      searchLabel,
      this.renderToggleButton()
    );
  }
});

export default connect(
  state => ({
    endPanelCollapsed: getPaneCollapse(state, "end")
  }),
  dispatch => bindActionCreators(actions, dispatch)
)(WelcomeBox);
