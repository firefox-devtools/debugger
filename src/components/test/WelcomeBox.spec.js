/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { shallow } from "enzyme";

import { WelcomeBox } from "../WelcomeBox";

describe("WelomeBox", () => {
  const setActiveSearch = () => null;

  it("renders with default values", () => {
    const wrapper = shallow(<WelcomeBox setActiveSearch={setActiveSearch} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("doesn't render toggle button in horizontal mode", () => {
    const horizontal = true;
    const wrapper = shallow(
      <WelcomeBox horizontal={horizontal} setActiveSearch={setActiveSearch} />
    );
    expect(wrapper.find("PaneToggleButton")).toHaveLength(0);
  });

  it("calls correct function on searchSources click", () => {
    const openQuickOpen = jest.fn();
    const wrapper = shallow(
      <WelcomeBox
        setActiveSearch={setActiveSearch}
        openQuickOpen={openQuickOpen}
      />
    );
    wrapper.find(".welcomebox__searchSources").simulate("click");
    expect(openQuickOpen).toBeCalled();
  });

  it("calls correct function on searchProject click", () => {
    const setActiveSearchSpy = jest.fn();
    const wrapper = shallow(
      <WelcomeBox setActiveSearch={setActiveSearchSpy} />
    );
    wrapper.find(".welcomebox__searchProject").simulate("click");
    expect(setActiveSearchSpy).toBeCalled();
  });
});
