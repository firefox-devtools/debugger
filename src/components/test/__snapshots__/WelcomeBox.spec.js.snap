// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelomeBox renders with default values 1`] = `
<div
  className="welcomebox"
>
  <div
    className="alignlabel"
  >
    <div
      className="shortcutFunction"
    >
      <p
        className="welcomebox__searchSources"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="shortcutKey"
        >
          Ctrl P
        </span>
        <span
          className="shortcutLabel"
        >
           to search for sources
        </span>
      </p>
      <p
        className="welcomebox__searchProject"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="shortcutKey"
        >
          Ctrl Shift F
        </span>
        <span
          className="shortcutLabel"
        >
           to find in files
        </span>
      </p>
    </div>
    <PaneToggleButton
      collapsed={true}
      handleClick={[MockFunction]}
      horizontal={false}
      position="end"
    />
  </div>
</div>
`;
